go/vt/vtgate/executor.go

```diff
@@ -307,7 +307,16 @@ func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql
 	qr, err := plan.Instructions.Execute(vcursor, bindVars, true)
 
 	logStats.ExecuteTime = time.Since(execStart)
-	queriesProcessed.Add(plan.Instructions.RouteType(), 1)
+
+	// TODO (@rafael): For now only report in the first ks/table in the primitive.
+	// Update this code to report all keyspaces/tables.
+	var tableName string
+	if len(plan.Instructions.KeyspaceTableNames()) > 0 {
```

Comment:
I think it's important to handle the multi-table (and multi-keyspace) case before merging... but I actually think the current implementation is more complex than it needs to be.

The only case in which we have more than one value for either keyspace or table is in Joins. So... instead of building up an array that you later have to join together, I think when getting the labels for a Join plan, it should concatenate the keyspace(s) and table(s) together into a single string each down in the Join code itself, so that the generic processing here can assume that it will just get a single label for each of the keyspace and the table name

Reply:
routes can also have multiple tables, but your comment applies equally well to them.